_posts/2024/2024-12-02-Cobbler-v3.3.7-Debian-Deployment-Guide.md

```diff
@@ -0,0 +1,423 @@
+
+```shell
+[ -e /mnt/Debian ] || mkdir /mnt/Debian
+mount -t iso9660 -o loop,ro /home/fedora/Downloads/debian-11.11.0-amd64-DVD-1.iso /mnt/Debian
+cobbler import --name=Debian11.11 --path=/mnt/Debian
+
+cd /var/www/cobbler/distro_mirror/Debian11.11/dists/bullseye
+gpg -u cobbler -bao Release.gpg Release
+gpg -u cobbler --clear-sign --output InRelease Release
+
+cat /var/www/cobbler/distro_mirror/Debian11.11/install.amd/initrd.gz ~/Downloads/Debian11.11-netboot.gz ~/Downloads/Debian11.11-firmware.gz > /var/www/cobbler/pub/Debian11.11-netboot.gz
+cobbler distro edit --name Debian11.11-x86_64 --initrd "/var/www/cobbler/pub/Debian11.11-netboot.gz"
+
+grep -v "os_prober" /var/lib/cobbler/templates/bookworm-sample.seed > /var/lib/cobbler/templates/bullseye-sample.seed
+cobbler profile edit --name Debian11.11-x86_64 --autoinstall bullseye-sample.seed
+
+cobbler system add --name Debian11.11 --profile Debian11.11-x86_64 --mac-address "aa:bb:cc:dd:ee:ff" --netboot-enabled true
+
+systemctl restart cobblerd && sleep 10
+cobbler sync
+```
+
+## Mirroring with debmirror
+
+Mirroring the official Debian repos with the **debmirror** tool is fairly simple, but will require about 100GB of extra disk space on the Cobbler server.
+
+This guide will only provide the steps to mirror the LATEST release of Debian *(Bookworm 12.8 at the time of writing)* with `debmirror`.
+
+similarly to the DVD/ISO method above, both the **netboot** and **DVD/ISO** `initrd.gz` images are still required in the same concatenated manner to properly install over the network, but they must both be downloaded separately of the `debmirror` process.
+
+Since the official Debian repo will be mirrored, the source will be fully valid with trusted signatures, meaning the autoinstall (preseed) file for the PXE client will not need the APT security settings required in the DVD/ISO source method.
+
+Cobbler supports managing repos/sources via the debmirror command natively, which means the `debmirror` package only needs to be installed and configured, then mirroring can be accomplished/managed through Cobbler CLI commands.
+
+- The syntax for mirroring a Debian repo through the `debmirror` command will still be provided below for context in the **Manual debmirror** section, but is not necessary in most cases to mirror Debian locally.
+
+### Setup debmirror
+
+Follow the instructions for supporting Debian deployments/repos given from the `cobbler check` command output (as well as installign Debian keyrings):
+
+```shell
+yum install -y debmirror debian-keyring
+sed -i "s/@arches/#@arches/g" /etc/debmirror.conf
+sed -i "s/@dists/#@dists/g" /etc/debmirror.conf
+systemctl restart cobblerd && sleep 5
+cobbler sync
+cobbler check
```

Comment:
Same here: I haven't tried it but I see no need for this restart. Restarting bigger cobbler instances can take minutes which is undesired I would argue.

```suggestion
yum install -y debmirror debian-keyring
sed -i "s/@arches/#@arches/g" /etc/debmirror.conf
sed -i "s/@dists/#@dists/g" /etc/debmirror.conf
cobbler sync
cobbler check
```